dataland-internal-storage/src/main/kotlin/db/migration/V25__UpdateSfdrModel.kt

```diff
@@ -0,0 +1,104 @@
+            "rateOfAccidentsInPercent" to "rateOfAccidents",
+        )
+
+    /**
+     * Migrates the rate of accidents in the sfdr data
+     */
+    fun migrateRateOfAccidents(dataTableEntity: DataTableEntity) {
+        val dataset = dataTableEntity.dataJsonObject
+
+        val socialObject = dataset.optJSONObject("social") ?: return
+        val socialAndEmployeeMattersObject = socialObject.optJSONObject("socialAndEmployeeMatters") ?: return
+
+        for ((oldKey, newKey) in oldToNewRateOfAccidentsKey) {
+            val oldValue = socialAndEmployeeMattersObject.remove(oldKey) ?: continue
+            socialAndEmployeeMattersObject.put(newKey, oldValue)
+        }
+
+        dataTableEntity.companyAssociatedData.put("data", dataset.toString())
+    }
+
+    /**
+     * Migrates the two fields excessiveCeoPayRatioInPercent and ceoToEmployeePayGapRatio into a single field
+     * called excessiveCeoPayRatio.
+     */
+    fun migrateExcessiveCeoPayGapRatio(dataTableEntity: DataTableEntity) {
+        val dataset = dataTableEntity.dataJsonObject
+
+        val socialObject = dataset.optJSONObject("social") ?: return
+        val socialAndEmployeeMattersObject = socialObject.optJSONObject("socialAndEmployeeMatters") ?: return
+
+        val excessiveCeoPayRatioInPercentValue = socialAndEmployeeMattersObject.remove("excessiveCeoPayRatioInPercent") as? JSONObject
+        val ceoToEmployeePayGapRatioValue = socialAndEmployeeMattersObject.remove("ceoToEmployeePayGapRatio") as? JSONObject
+        val valueToUse = determineValueToUse(excessiveCeoPayRatioInPercentValue, ceoToEmployeePayGapRatioValue)
+
+        valueToUse?.let { socialAndEmployeeMattersObject.put("excessiveCeoPayRatio", it) }
+
+        if (valueToUse != null) {
+            dataTableEntity.companyAssociatedData.put("data", dataset.toString())
+        }
+    }
+
+    /**
+     * Determine which value to use put into the new excessiveCeoPayRatio field
+     */
+    private fun determineValueToUse(
+        excessiveCeoPayRatioInPercentValue: JSONObject?,
+        ceoToEmployeePayGapRatioValue: JSONObject?,
+    ): JSONObject? =
+        when {
+            excessiveCeoPayRatioInPercentValue != null && ceoToEmployeePayGapRatioValue == null -> excessiveCeoPayRatioInPercentValue
+            excessiveCeoPayRatioInPercentValue == null && ceoToEmployeePayGapRatioValue != null -> ceoToEmployeePayGapRatioValue
+            excessiveCeoPayRatioInPercentValue != null && ceoToEmployeePayGapRatioValue != null -> {
+                when {
+                    isValueOfObjectAValidNumber(ceoToEmployeePayGapRatioValue) -> ceoToEmployeePayGapRatioValue
+                    isValueOfObjectAValidNumber(excessiveCeoPayRatioInPercentValue) -> excessiveCeoPayRatioInPercentValue
+                    else -> null
+                }
+            }
+            else -> null
+        }
+
+    /**
+     * Checks if the given value is a number
+     */
+    fun isNumber(value: Any?): Boolean =
```

Comment:
```suggestion
    private fun isNumber(value: Any?): Boolean =
```

Also, I think this should be easier. Suggestion: Try to modify the `isValueOfObjectAValidNumber` function in the following way:

```suggestion
    private fun isValueOfObjectAValidNumber(jsonObject: JSONObject): Boolean = jsonObject.has("value") && jsonObject.get("value").toString().toDoubleOrNull() != null 
```
and see if that works.

Reply:
done